### PR TITLE
Simple web server for testing ingress-gce features

### PR DESCRIPTION
**Pull request by nicksardo**

### CODE REVIEW

File: cmd/echo/app/handlers.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	requestDump, err := httputil.DumpRequest(r, true)
+	if err != nil {
+		glog.Error(err)
+	} else {
+		v := string(requestDump)
+		w.Write([]byte(html.EscapeString(v)))
+	}
+
+	w.Write([]byte(fmt.Sprintf("remote-address: %v\n", r.RemoteAddr)))
```
Comment (bowei) [inline cmd/echo/app/handlers.go]: Can we put this in a JSON and make it easier to parse?

File: cmd/echo/app/tls.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+func createCert() (string, string) {
```
Comment (bowei) [inline cmd/echo/app/tls.go]: docstring

File: cmd/echo/app/handlers.go (hunk 1 above)
Comment (nicksardo) [reply cmd/echo/app/handlers.go]: Problem is that there's only value in JSON if the entire body is JSON. The httputil dumper only returns a []byte, so we have no easy way of incorporating that into a JSON struct without killing the readability.
```
GET / HTTP/1.1
Host: localhost:8080
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Cache-Control: max-age=0
Connection: keep-alive
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.62 Safari/537.36

{
	"env": {
		"Node": "",
		"Pod": "",
		"Namespace": ""
	},
	"remoteAddr": "[::1]:46372",
	"tls": false,
	"dump": "GET / HTTP/1.1\r\nHost: localhost:8080\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8\r\nAccept-Encoding: gzip, deflate, br\r\nAccept-Language: en-US,en;q=0.9\r\nCache-Control: max-age=0\r\nConnection: keep-alive\r\nUpgrade-Insecure-Requests: 1\r\nUser-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.62 Safari/537.36\r\n\r\n"
}
```

Comment (nicksardo) [reply cmd/echo/app/handlers.go]: Will stop using httputil dump and hand-roll it:
```
{
	"requestMethod": "GET",
	"requestURI": "/",
	"protocol": "HTTP/1.1",
	"k8sEnv": {
		"Node": "",
		"Pod": "",
		"Namespace": ""
	},
	"remoteAddr": "[::1]:52506",
	"tls": false,
	"dump": "",
	"header": {
		"Accept": [
			"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8"
		],
		"Accept-Encoding": [
			"gzip, deflate, br"
		],
		"Accept-Language": [
			"en-US,en;q=0.9"
		],
		"Connection": [
			"keep-alive"
		],
		"Upgrade-Insecure-Requests": [
			"1"
		],
		"User-Agent": [
			"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.62 Safari/537.36"
		]
	}
}
```

**Pull request merged**